src/Client.js

```diff
@@ -11,22 +12,19 @@ function prettyPrintResults(results) {
 export default class Client {
   constructor(port) {
     this.port = port;
-    this.completedFullRun = false;
   }
 
   connect() {
     const d = dnode.connect(this.port);
     d.on('remote', function(remote) {
       setInterval(() => {
         remote.status('', results => {
-          // TODO(allenk): make the results more robust, and invalidate full runs from the server
-          if (results.length && !this.completedFullRun) {
-            this.completedFullRun = true;
+          if (!results.message) {
             prettyPrintResults(results);
-            d.end();
-            process.exit(0);
+            d.close();
+            process.exit(0)
           } else {
-            return;
+            console.log(results.message, results.files);
```

Comment:
eventually we can replace this with something that overwrites the previous line (for a nice CLI interface)